extensions/register.md

```diff
@@ -0,0 +1,157 @@
+---
+title: The REGISTER command
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Ed Kellett"
+    period: "2020"
+    email: "e@kellett.im"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+
+## Motivation
+
+The lack of a standardized way to register an account has been a source of frustration for many participants
+for several years.
+
+This isn't the first attempt to fix it. Unlike the withdrawn `ACC` proposal, this effort aims to be more
+appealing to implementors by being severely restricted in scope. The only register-verify flows supported
+are those already in use in the wild. The authors hope that this stripped-down solution will serve as the
+basis for future organic expansion.
+
+### Capability
+
+This specification adds the `draft/register` capability, whose presence signifies that the server
+accepts the `REGISTER` command.
+
+The capability has an optional value, a comma-separated list of key-value pairs; the format is intended
+to follow the precedent set by [the multiline draft][multiline].
+
+The defined keys are:
+
+ * `before-connect` - if present, indicates the server supports early registration, so it
+   MUST NOT use `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
+ * `email-required` - if present, registrations require a valid email address to process
+
+Clients MUST ignore any value assigned to these keys, and MUST ignore any unknown key.
+
+### Commands
+
+    REGISTER {<email> | "*"} <password>
+    
+The `REGISTER` command informs the server of a request to register an account named for the current
+nick of the requestor.
+
+The `REGISTER` command MAY be sent at any point during the connection that the client has a valid
+nickname. If the client sends `REGISTER` before completing connection registration, and receives a
+`FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`, it SHOULD make a second attempt after it receives the
+welcome message.
+
+    VERIFY <account> <code>
+    
+The `VERIFY` command completes a registration that required email verification.
+
+### Responses
+
+    REGISTER SUCCESS <account> <message>
+    VERIFY SUCCESS <account> <message>
+    
+Sent by the server when the `REGISTER` or `VERIFY` fully succeeded, and no
+further action is needed.
+Afterward, the client is assumed to be authenticated to the `<account>`
+as if it used SASL.
+
+    REGISTER VERIFICATION_REQUIRED <account> <message>
+    
+Sent by the server as a response to `REGISTER` when further action is required
+to complete registration of the `<account>`, as explain in the human-readable `<message>`.
+
+    FAIL REGISTER USERNAME_EXISTS <account> <message>
```

Comment:
```suggestion
    FAIL REGISTER ACCOUNT_EXISTS <account> <message>
```

Would prevent overloaded terminology (username can also refer to what is also known as the ident).